### PR TITLE
EXLM-1922 Refactored the signup flow code

### PR DESCRIPTION
**Pull request by mayukhs2904**
Please always provide the Jira Issue your PR is for, as well as test URLs where your change can be observed (before and after):

Jira ID: https://jira.corp.adobe.com/browse/EXLM-1922

Test URLs:

- Before: https://main--exlm--adobe-experience-league.hlx.page
- After: https://feat-exlm-1922--exlm--adobe-experience-league.hlx.page

### CODE REVIEW

File: scripts/profile/profile.js [hunk 1]
```diff
@@ -1,142 +1,166 @@
...
+    email: profileData?.email || '',
+    industry: profileData?.industryInterests || '',
+    roles: profileData?.role || [],
+    interests: profileData?.interests || [],
+    profilePicture: ppsProfileData?.images?.['100'] || '',
+    company: ppsProfileData?.company || '',
+    communityUserName: communityProfileDetails?.username || '',
+    communityUserTitle: communityProfileDetails?.title || '',
+    communityUserLocation: communityProfileDetails?.location || '',
+  };
```
Comment (noormd08) [inline scripts/profile/profile.js:37]: Can we add the restriction here as well?

File: scripts/profile/profile.js [hunk 2]
```diff
@@ -1,142 +1,166 @@
...
+                }: </span><span>${industry}</span></div>`
+              : ''
+          }
+          ${
+            interests &&
+            ((Array.isArray(interests) && interests.length > 0) ||
+              (typeof interests === 'string' && interests.trim() !== ''))
+              ? `<div class="user-interests"><span class="heading">${
+                  placeholders?.myInterests || 'My Interests'
+                }: </span><span>${interests.join('&nbsp;&nbsp;')}</span></div>`
```
Comment (noormd08) [inline scripts/profile/profile.js:154]: I think, we need to join with comma not space. Can you please check the XD once.

File: scripts/profile/profile.js [hunk 3]
```diff
@@ -1,142 +1,166 @@
...
+
+  const additionalProfileInfoDOM = `
+    <div class="profile-row additional-data">
+      <div class="profile-card-body additional-data-body">
+        <div class="profile-user-info">
+          ${
+            roles && ((Array.isArray(roles) && roles.length > 0) || (typeof roles === 'string' && roles.trim() !== ''))
+              ? `<div class="user-role"><span class="heading">${
+                  placeholders?.myRole || 'My Role'
+                }: </span><span>${roles.join('&nbsp;&nbsp;')}</span></div>`
```
Comment (noormd08) [inline scripts/profile/profile.js:136]: I think, we need to join with comma not space. Can you please check the XD once.

Comment (mayukhs2904) [reply scripts/profile/profile.js:136]: it's space, verified it with XD.

**Review by noormd08 — APPROVED** [review]

**Pull request merged**